exercises/concept/inventory-management/.docs/introduction.md

```diff
@@ -0,0 +1,186 @@
+```
+
+#### One-operand form
+
+Two registers are implicitly used to perform a multiplication in one-operand form: `rax` and `rdx`.
+If the multiplication involves two 64-bit numbers, then the lower 64-bits of the result will be in `rax` and the upper 64-bits will be in `rdx`.
+
+This is usually called `rdx:rax`, to indicate that both registers are used in tandem.
+
+The same happens if the multiplication involves different register sizes.
+So, for instance, if two 32-bit numbers are being multiplied, `eax` and `edx` will be used.
+
+The exception is the multiplication between two bytes.
+
+In this case, instead of `dl:al`, `ax` will be used.
+The lower portion of `ax` (`al`) will get the lower 8-bits of the product, while the upper portion (`ah`) will get the upper 8-bits.
+
+~~~~exercism/caution
+The `rdx` register is implicitly used in an one-operand multiplication.
+This means any necessary value in `rdx` must be saved before that operation.
+~~~~
+
+#### Two-operand form
+
+The two-operand form of `imul` allows for explicitly declaring a different destination operand.
+
+In that case, the product between source and destination operands is placed in the destination operand.
+This product is truncated to fit into the destination operand.
+
+#### Three-operand form
+
+The three-operand form of `imul` multiplies the two source operands and places the result in the destination operand.
+
+This means that the destination operand is not multiplied with any of the source operands, it just receives the result.
+This product is also truncated to fit into the destination operand.
+
+#### Handling overflow
+
+In case of a possible overflow, it is sometimes useful to move operands to a larger register size.
+
+A `movzx` instruction can be used to convert a value in a 8-bits or 16-bits source operand to a larger destination operand, clearing all remaining bits (this is called `zero extension`):
+
+```nasm
+mov ax, 1000 ; lower 16-bits of eax are 1000, upper bits are undefined
+mov cx, 200 ; lower 16-bits of ecx are 200, upper bits are undefined
+
+; 200 * 1000 does not fit in 16-bits, so a 32-bits multiplication is necessary
+; however, multiplying eax by ecx may produce an incorrect result due to undefined bits
+
+movzx eax, ax ; lower 16-bits of eax remain 1000, upper bits are cleared
+movzx ecx, cx ; lower 16-bits of ecx remain 200, upper bits are cleared
+mul ecx ; now eax correctly holds 200 * 100
+```
+
+A 32-bits source operand is always zero-extended to all 64-bits of the destination operand with a simple `mov`.
+
+#### Using imul with unsigned numbers
+
+It is possible to use `imul` to multiply unsigned numbers.
+However, this may give an incorrect result if one of the numbers may be interpreted as negative.
+
+Since the `sign bit` corresponds to the most significant bit in an unsigned integer, in practice, this difference becomes relevant when more than the truncated result is needed, ie, when the full range of `rdx:rax` is used.
```

Comment:
I do not think that `sign bit` is what is meant here, but _sign bit_ emphasis could be used.  It probably should not be code delimited.